sympy/polys/rationaltools.py

```diff
@@ -83,3 +83,68 @@ def _together(expr):
         return expr
 
     return _together(sympify(expr))
+
+
+def thiele_interpolate(u, v, var=symbols('x'), simplify=True):
+    """
+    Build a rational function from a finite set of inputs in vector u
+    and their function values in vector v (both vectors must be of equal
+    lengths).
+
+    This algorithm might encounter division by zero
```

Comment:
What is an example of this (or reference)?

Reply:
A ZeroDivisionError is yielded whenever values are equally spaced. For instance,
`thiele([0,1,2,3], [0,1,2,3])`

Reply:
I would like to see a better criteria stated. If you interpolate on `([1+z,2,3,4],(1,2,3,5))` and then replace `z` with 0 you get `x` which is clearly not right. If you use only 3 points as in `(1,2,3),(1,2,3)` it doesn't fail. Maybe this should be limited to numeric input whose output is unambiguously ok or is an error indicating problems.

Reply:
Is the criteria that there cannot be 3 colinear points? There is a routine for checking colinearity in geometry.